src/main/java/net/dries007/tfc/api/recipes/LoomRecipe.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.api.recipes;
+
+import net.dries007.tfc.api.registries.TFCRegistries;
+import net.minecraft.item.Item;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+import javax.annotation.Nullable;
+
+public class LoomRecipe extends IForgeRegistryEntry.Impl<LoomRecipe>
+{
+    @Nullable
+    public static LoomRecipe get(Item item)
+    {
+        return TFCRegistries.LOOM.getValuesCollection().stream().filter(x -> x.isValidInput(item)).findFirst().orElse(null);
+    }
+
+    @Nullable
+    public static LoomRecipe getByOutput(Item item)
+    {
+        return TFCRegistries.LOOM.getValuesCollection().stream().filter(x -> x.isValidOutput(item)).findFirst().orElse(null);
+    }
+
+    private Item inputItem;
```

Comment:
This should use `IIngredient<ItemStack>` instead of just an `Item`. As it is right now, it doesn't support metadata comparisons, or ore dictionary values. See `BarrelRecipe` for an example

Reply:
Okay, I'm gonna fix that